File: app/src/main/java/com/ronju/covid_19tracker/Activitys/Fragment/CountryViewActivity.java
package com.ronju.covid_19tracker.Activitys.Fragment;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.app.Dialog;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.FrameLayout;
import android.widget.TextView;

import com.ronju.covid_19tracker.Adapter.CountryAdapter;
import com.ronju.covid_19tracker.LoadingDialog;
import com.ronju.covid_19tracker.R;

public class CountryViewActivity extends Fragment {
    private RecyclerView recyclerView;
    private CountryAdapter mAdapter;
    private Dialog loadingDialog;
    private androidx.appcompat.widget.SearchView searchView;
    private View view = null;
    private FrameLayout frameLayout;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        view = inflater.inflate(R.layout.activity_country_view, container, false);
        loadingDialog = LoadingDialog.getCustomLoadingDialog(getContext());
        ((AppCompatActivity)getActivity()).getSupportActionBar().setTitle("Choose your country");
        frameLayout = view.findViewById(R.id.fragmentViewer);

        intView();


        LinearLayoutManager llm = new LinearLayoutManager(getContext());
        llm.setOrientation(LinearLayoutManager.VERTICAL);
        recyclerView.setLayoutManager(llm);
        recyclerView.setHasFixedSize(true);
        mAdapter = new CountryAdapter(getContext());
        recyclerView.setAdapter(mAdapter);


        searchView.setOnQueryTextListener(new androidx.appcompat.widget.SearchView.OnQueryTextListener() {
            @Override
            public boolean onQueryTextSubmit(String query) {
                return false;
            }

            @Override
            public boolean onQueryTextChange(String newText) {
                mAdapter.getFilter().filter(newText);
                return false;
            }
        });



        return view;
    }


    private void intView() {
        recyclerView = view.findViewById(R.id.countryItemView);
        searchView = view.findViewById(R.id.cSearch);
    }


}

File: app/src/main/java/com/corona/covid_19tracker/Activity/Fragment/CountryViewActivity.kt
package com.corona.covid_19tracker.Activity.Fragment

import com.corona.covid_19tracker.Adapter.CountryAdapter
import android.view.LayoutInflater
import android.view.ViewGroup
import android.os.Bundle
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import androidx.recyclerview.widget.LinearLayoutManager
import com.corona.covid_19tracker.Api.CountryDataRetrofitHelper
import com.corona.covid_19tracker.Api.CountryDataService
import com.corona.covid_19tracker.Model.CountryDataModel
import com.corona.covid_19tracker.Repository.CountryRepository
import com.corona.covid_19tracker.Repository.Response
import com.corona.covid_19tracker.ViewModel.CountryViewModel
import com.corona.covid_19tracker.ViewModel.CountryViewModelFactory
import com.corona.covid_19tracker.databinding.ActivityCountryViewBinding
import com.corona.covid_19tracker.utils.AdsTask

class CountryViewActivity : Fragment() {

    private lateinit var binding: ActivityCountryViewBinding
    private var mAdapter: CountryAdapter? = null
    private var counter = 0

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        binding = ActivityCountryViewBinding.inflate(layoutInflater)
        (activity as AppCompatActivity?)!!.supportActionBar!!.title = "Choose your country"
        val countryDataService =
            CountryDataRetrofitHelper.getInstance().create(CountryDataService::class.java)
        val repository = CountryRepository(countryDataService)
        val viewModel = ViewModelProvider(
            requireActivity(),
            CountryViewModelFactory(repository)
        ).get(CountryViewModel::class.java)

        viewModel.result.observe(requireActivity(), Observer {
            when (it) {
                is Response.Loading -> {
                    binding.countryShimmer.visibility = View.VISIBLE
                    binding.countryShimmer.startShimmer()
                }
                is Response.Success -> {
                    binding.countryShimmer.startShimmer()
                    binding.countryShimmer.visibility = View.INVISIBLE
                    it.data?.let {
                        if (it.size != 0) {
                            setAdapter(it)
                        }
                    }
                }
                is Response.Error -> {
                    viewModel.getResult()
                    if (counter == 0)
                        Toast.makeText(context, it.errorMessage, Toast.LENGTH_LONG).show()

                }
            }

        })



        binding.cSearch.setOnQueryTextListener(
            object :
                androidx.appcompat.widget.SearchView.OnQueryTextListener {
                override fun onQueryTextSubmit(query: String?): Boolean {
                    return false
                }

                override fun onQueryTextChange(newText: String?): Boolean {
                    mAdapter?.filter?.filter(newText)
                    return false
                }

            })
        AdsTask(requireContext()).setBannerAds(binding.adView)
        return binding.root
    }

    private fun setAdapter(countryDataModel: CountryDataModel) {
        binding.countryItemView.layoutManager =
            LinearLayoutManager(context, LinearLayoutManager.VERTICAL, false)
        mAdapter = CountryAdapter(countryDataModel)
        binding.countryItemView.adapter = mAdapter
        binding.countryItemView.scheduleLayoutAnimation()
    }
}